venues/NeurIPS.cc/2020/Conference/python/get-reviewer-expertise.py

```diff
@@ -0,0 +1,48 @@
+import openreview
+import argparse
+import csv
+from tqdm import tqdm
+
+def get_publications(openreview_client, author_id):
+    content = {
+        'authorids': author_id
+    }
+    publications = openreview.tools.iterget_notes(openreview_client, content=content)
+    return [publication for publication in publications]
+
+if __name__ == '__main__':
+    ## Argument handling
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--baseurl', help="base url")
+    parser.add_argument('--username')
+    parser.add_argument('--password')
+    args = parser.parse_args()
+
+    client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+    infile = 'neurips20_program_committee_new.csv'
+    outfile = 'neurips20_email_status.csv'
+
+    with open(infile, 'r') as f, open(outfile, 'w') as f2:
+        csv_reader = csv.reader(f)
+        next(csv_reader)
+        map_email_to_details = {}
+        for line in tqdm(csv_reader):
+            email = line[0].strip().lower()
+            map_email_to_details[email] = {
+                'fname': line[1].strip(),
+                'lname': line[2].strip()}
+        map_profiles = client.search_profiles(emails=list(map_email_to_details.keys()))
+        map_profiles_done = {}
+        csv_writer = csv.writer(f2)
+        csv_writer.writerow(['email', 'firstName', 'lastName', 'profile found', 'active', 'count of publications'])
+        for email, content in tqdm(map_email_to_details.items()):
+            profile = map_profiles.get(email)
+            if profile:
+                if profile.id in map_profiles_done:
+                    continue
```

Comment:
I'm seeing a difference of 2 lines between the input and output files, do you know why?

why do we have a `continue` here?

Reply:
I am aggregating profiles from emails into this dict `map_profiles_done`. I am guessing the missing 2 rows belong to emails that have other emails belonging to the same profile and therefore are already there in the `map_profiles_done`. There is no strong reason for having the `continue`. I can modify it to :
`if profile.id not in map_profiles_done: map_profiles_done[profile.id] = email`.

Reply:
I'm not sure if we should tell them there are two repeated profiles or show the profiles twice, I want to have in and out consistent.

Reply:
Ok, I have made some changes and now in and out should remain consistent. I am now additionally printing out any duplicates observed while writing the output.